### PR TITLE
Library Search : New event for list of providers, adjusted filter.

### PR DESCRIPTION
**Pull request by PeterMis**
- New event that returns an array of enabled library providers with their details (Pixabay as it stands)
- Adjusted librarySearch `provider` filter to expect `local`, `pixabay` etc

### CODE REVIEW

File: lib/Controller/Library.php [hunk 1]
```diff
@@ -876,39 +878,66 @@ public function search(Request $request, Response $response): Response
...
-
-            // Do we have a type filter
-            $types = $parsedQueryParams->getArray('types');
-            $type = $parsedQueryParams->getString('type');
-            if ($type !== null) {
-                $types[] = $type;
-            }
+        } else {
+            // Dispatch event to get Providers and their details.
+            $eventList = new LibraryProviderListEvent();
```
[the remark this answers is not in the document]
Comment (PeterMis) [reply lib/Controller/Library.php:883]: Hmm, yeah I suppose we could just dispatch the `LibraryProviderEvent` with the provider 🤔 
I'm not entirely sure why did this like that, let me remove the other event and some stuff

[the remark this answers is not in the document]
Comment (PeterMis) [reply lib/Controller/Library.php]: Updated, Removed the providersList event and the foreach/if, as we indeed don't need to do that there I don't think

**Review by dasgarner — APPROVED** [review]
Approved, please check with Mauro before merging in case he wants to pull it in first.

**Comment by PeterMis** [comment]
Merged as a part of this PR - https://github.com/xibosignage/xibo-cms/pull/2398
As such I will close this one.

**Pull request closed without merging**